### PR TITLE
Support setting the XHR responseType

### PR DESCRIPTION
**Pull request by mhahn**
Proposed solution to #654

### CODE REVIEW

**Comment by rylan** [comment]
+1, having this feature would be super handy

**Comment by deltaidea** [comment]
:+1:

**Comment by ponelat** [comment]
+1

File: lib/client.js [hunk 1]
```diff
@@ -880,6 +880,17 @@ Request.prototype.withCredentials = function(){
 };
 
 /**
+ * Enable setting the XHR responseType.
+ *
+ * @api public
+ */
+
+Request.prototype.responseType = function(type){
+    this._xhrResponseType = type;
```
Comment (rjonna) [inline lib/client.js]: You will have to use xhr.response instead of xhr.responseText at line 300.

**Comment by jbq** [comment]
Works great, thanks!  I had to manually resolve conflicts however, this pull request has to be updated.  See https://github.com/jbq/superagent/commit/7a8ddf4ac193573b608e592ab3a6a436eb144f0d

**Comment by brianswisher** [comment]
:+1:

File: lib/client.js (hunk 1 above)
Comment (mhahn) [inline lib/client.js]: to reference the response you should do: response.xhr.response

**Comment by mhahn** [comment]
@pornel just rebased

**Comment by pornel** [comment]
The tests are failing: Error: InvalidStateError: DOM Exception 11: An attempt was made to use an object that is not, or is no longer, usable. (https://jmxihwatjo.localtunnel.me/__zuul/test-bundle.js:509)

**Comment by mhahn** [comment]
hm, they're passing locally for me

**Comment by mhahn** [comment]
is there a way i can repro that locally? i can't even see which test failed

**Comment by pornel** [comment]
It looks like this code is not compatible with internet explorer 10 on Windows 2012:

https://travis-ci.org/visionmedia/superagent/jobs/96181019

it passes in other browsers.

**Comment by focusaurus** [comment]
Not sure the exact history of things here, but a `responseType(name)` method has made it into superagent releases. OK to close this out as resolved/duplicate?

**Pull request closed without merging**